tests/Public/New-GuestConfigurationPolicy.Tests.ps1

```diff
@@ -690,6 +690,75 @@ Describe 'New-GuestConfigurationPolicy' {
                         Assert-GuestConfigurationPolicyDefinitionFileValid @assertionParameters
                     }
                 }
+
+                Context 'Optional identity parameters - ManagedIdentityResourceId and LocalContentPath, ExcludeArcMachines' {
+                    BeforeAll {
+                        $fileName = $ContentUri.Split('/')[-1]
+                        $filePath = Join-Path -Path $defaultDefinitionsPath -ChildPath $fileName
+                        Invoke-WebRequest $ContentUri -OutFile $filePath
+
+                        $basePolicyParameters['ManagedIdentityResourceId'] = 'myManagedIdentity'
+                        $basePolicyParameters['LocalContentPath'] = $filePath
+                        $baseAssertionParameters['ExpectedManagedIdentity'] = 'myManagedIdentity'
+                    }
+
+                    It 'Should include contentManagedIdentity in the result object and exclude Arc machines' {
+                        $result = New-GuestConfigurationPolicy @basePolicyParameters -ExcludeArcMachines
+
+                        $result | Should -Not -BeNull
+
+                        $fileContent = Get-Content -Path $result.Path -Raw
+                        $fileContentJson = $fileContent | ConvertFrom-Json
+
+                        $fileContentJson.properties.metadata.guestConfiguration.contentManagedIdentity | Should -Be 'myManagedIdentity'
+
+                        # Check Hybrid section removed
+                        $imageConditionList = $fileContentJson.properties.policyRule.if.anyOf
+
+                        $imageConditionList[0].allOf[0].anyOf[0].equals | Should -Be 'Microsoft.Compute/virtualMachines'
+                        $imageConditionList[1].allOf | Should -BeNullOrEmpty
+                    }
+
+                    It 'Should not include contentManagedIdentity in the result object, but also exclude Arc machines' {
+                        $basePolicyParameters.Remove('ManagedIdentityResourceId')
+                        $basePolicyParameters.Remove('LocalContentPath')
+                        $result = New-GuestConfigurationPolicy @basePolicyParameters -ExcludeArcMachines
+
+                        $result | Should -Not -BeNull
+
+                        $fileContent = Get-Content -Path $result.Path -Raw
+                        $fileContentJson = $fileContent | ConvertFrom-Json
+
+                        $fileContentJson.properties.metadata.guestConfiguration.contentManagedIdentity | Should -BeNullOrEmpty
```

Comment:
I think this check isn't quite enough. You want to make sure that the key doesn't even exist, not that the key exists, but its value is empty?